### PR TITLE
Run example tests in CI (and fix a test)

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

Run the tests for our example contracts in our CI to ensure that they work.
Fix the WCCD contract such that its tests pass.

## Changes

- Add a new CI job
- Use the correct length for transfers in the WCCD contract

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -274,3 +274,46 @@ jobs:
           command: test
           # only run the library tests, no doc tests
           args: --manifest-path ${{ matrix.lib-crates }} --target=${{ matrix.target }} --features=${{ matrix.features }} --lib
+
+  test-examples:
+    name: Test examples
+    runs-on: ubuntu-latest
+    needs: rustfmt
```
Comment (abizjak) [inline .github/workflows/linter.yml:281]: I think this should also require all of the clippy and checks.

**Review by abizjak — APPROVED** [review]
In principle this is OK, but the issue is, and part of the reason why it was not done yet, that we really should be running tests with cargo-concordium for this, not tests on x86.

But I have no direct objection to these changes, they just don't address the issue fully.

File: .github/workflows/linter.yml (hunk 1 above)
Comment (Bargsteen) [reply .github/workflows/linter.yml:281]: I could add *all* the examples to the `clippy-schema` job (which runs on wasm). Would that be sufficient or what do you mean by "checks"?

Comment (abizjak) [reply .github/workflows/linter.yml:281]: Sorry, I meant that the test job should only run after the clippy job. So to add to the `needs` field.

Comment (Bargsteen) [reply .github/workflows/linter.yml:281]: Ah, I see. I'll update it.

**Review by abizjak — DISMISSED** [review]

**Review by abizjak — APPROVED** [review]

**Pull request merged**